src/components/ArticlesComponents/ArticleProvider.jsx

```diff
@@ -0,0 +1,53 @@
+import { useParams } from "react-router-dom";
+import { useEffect, useState } from "react";
+import InfoCard from "./InfoCard";
+import CommentsPanel from "./CommentsPanel";
+
+const ArticleProvider = () => {
+  const { article_id } = useParams();
+  const [articleState, setArticle] = useState([{}]);
+  const [comments, setComments] = useState([{}]);
+
+  const fetchComments = () => {
+    fetch(
+      `https://news-api-ryanfoo.herokuapp.com/api/articles/${article_id}/comments`
+    ).then((res) => {
```

Comment:
I'd suggest exporting this functionality into an api file for all requests to tidy up your components